### DISCUSSION TITLE
Is there any reason to use external cms?

### DISCUSSION THREAD

**Discussion by nxy7** [medusajs/medusa#2681] [Feature Requests]
Hello, I'm configuring my medusa setup and tbh I don't really know why do headless ecommerces use external cms solutions for item descriptions. My understanding is that Medusa provides only short description and full product page should be stored with external cms. Is there any advantage to that? To me it seems like additional postgres table would avoid syncing problems, decrease deployment complexity, use much less resources etc etc. I'm sure there are reasons why it's done the way it is so here are my questions:
1) Why don't headless ecommerces (medusa included) don't have any way for storing long product descriptions and other product related content?
2) Are there any plugins that add that functionality?

Thanks for any responses, I'm genuinely trying to learn more about Medusa :-)

**Comment by srindom** [comment]
In the end, it is your call to structure your stack as you see fit, and nothing restricts you from adding CMS functionality to Medusa. To do so, you can:

- [Add a custom entity to your Medusa project](https://docs.medusajs.com/advanced/backend/entities/) (e.g., `ProductContent`). The entity would have a reference to the core Product and the fields you would want to add. 
- [Add custom endpoints](https://docs.medusajs.com/advanced/backend/endpoints/add) to query and manage the fields.
- Extend the Admin dashboard to add UI support for managing these fields.

You will now be able to use the custom data in your storefront.

The reason for not including this in Medusa by default is that we think there are excellent CMS systems out there that do a much better job of managing content than what we can reasonably do. At the same time, using an external CMS decouples your operational commerce data from the representation layer, giving more flexibility in how you organize internal teams, build customer experiences, and evolve your tech stack as you grow. Considerations like these are typically not relevant if you are just getting started, and in such cases, it may be perfectly fine to use a simple approach that holds all your content within Medusa.

Hope this is helpful :)

**Reply by nxy7** [reply]
Okay so if I understand correctly Medusa doesn't want to have too much opinions on the way you structure your app so it's more flexible for bigger teams. Totally understandable, thank you for clarifying.
